tutorials/dapt-curation/code/utils.py

```diff
@@ -0,0 +1,389 @@
+            start_idx = doc["text"].find("/*")
+            end_idx = doc["text"].find("*/")
+            if start_idx != -1 and end_idx != -1:
+                comment = doc["text"][start_idx : end_idx + 2]
+
+                # Create a Dask DataFrame from the given data
+                comment_df = dd.from_pandas(
+                    pd.DataFrame([{"text": comment}]), npartitions=1
+                )
+                # Create a DocumentDataset instance
+                comment_doc_dataset = DocumentDataset(comment_df)
+                redacted_comment = redact_pii(comment_doc_dataset)
+
+                # Extract the "text" column from the DocumentDataset
+                text_column = redacted_comment.df["text"]
+
+                # Convert the Dask Series to a Pandas Series (if needed)
+                text_series = text_column.compute()
+
+                # Get the first (and presumably only) value from the Series
+                text_value = text_series.iloc[0]
+
+                # Convert the value to a string
+                redacted_text_string = str(text_value)
+
+                doc["text"] = doc["text"].replace(comment, redacted_text_string)
+
+                # print("final", doc["text"])
+
+    # Serialize the updated dictionary back to a JSON string
+    updated_json_data = json.dumps(data, indent=2)
+    return updated_json_data
+
+
+def redact(dataset: DocumentDataset) -> DocumentDataset:
+    """
+    Redact PII on comment lines on a  given code dataset.
+
+    Args:
+        dataset (DocumentDataset): The dataset to be redacted.
+
+    Returns:
+        DocumentDataset: The redacted dataset.
+    """
+    # Extract relevant data from the Dask DataFrame
+    length = len(dataset)
+    data_dict = {
+        "documents": dataset.df.compute().to_dict(orient="records"),
+    }
+    # Serialize the dictionary to a JSON string
+    json_data = json.dumps(data_dict, indent=2)
+    redactd_json_data = comment_redaction(json_data)
+
+    # Deserialize the JSON string to a dictionary
+    redactd_data_dict = json.loads(redactd_json_data)
+
+    # Create a DataFrame from the dictionary
+    redactd_dataset_df = dd.from_pandas(
+        pd.DataFrame(redactd_data_dict["documents"]), npartitions=length
+    )
+
+    # Create a DocumentDataset instance
+    redactd_document_dataset = DocumentDataset(redactd_dataset_df)
+
+    return redactd_document_dataset
+
+
+def redact_pii(dataset: DocumentDataset) -> DocumentDataset:
+    """
+    Redacts personally identifiable information (PII) from a given dataset.
+
+    Args:
+        dataset (DocumentDataset): The dataset containing documents with PII.
+
+    Returns:
+        DocumentDataset: The redacted dataset with PII replaced by a generic value.
+    """
+    DEFAULT_MAX_DOC_SIZE = 3000000
+    redactor = Modify(
+        PiiModifier(
+            supported_entities=[
+                "PERSON",
+                "EMAIL_ADDRESS",
+                #                                "URL",
```

Comment:
Either uncomment, or remove entirely.

Reply:
Removed it, but it shows the developer the other possible options as well

Reply:
Thanks.
The usage is already documented and shown in other tutorials.